backend/infrahub/graphql/mutations/main.py

```diff
@@ -144,7 +145,33 @@ async def mutate_create(
     ) -> tuple[Node, Self]:
         context: GraphqlContext = info.context
         db = database or context.db
-        obj = await cls.mutate_create_object(data=data, db=db, branch=branch)
+
+        # TODO clarify whether we want to lock on any ancestor kind having uniqueness_constraint,
```

Comment:
We need to have the lock on the node AND all generics with uniqueness constraints.
Technically, if the uniqueness constraint is the same on the node and on a generic, it probably means that it was inherited from the generic and as such we could only apply the lock on the generic.

Reply:
I think it would be good to move the logic to extract the locks that are required for a given object to its own method because we'll need it in a few places, the first one that comes to mind is the `update` mutation